secure_message/application.py

```diff
@@ -93,24 +98,18 @@ def handle_exception(error):  # NOQA pylint:disable=unused-variable
     return app
 
 
-def cache_client_token(app):
-    token = get_client_token(app.config['CLIENT_ID'],
-                             app.config['CLIENT_SECRET'],
-                             app.config['UAA_URL'])
+def refresh_client_token_if_required(app):
+    if app.token_expires_at <= maya.now():
+        cache_client_token(app)
 
-    r = redis.StrictRedis(host=app.config['REDIS_HOST'],
-                          port=app.config['REDIS_PORT'],
-                          db=app.config['REDIS_DB'])
-    put_token(r, token)
 
+def cache_client_token(app):
+    app.token = get_client_token(app.config['CLIENT_ID'],
```

Comment:
I think this variable would be better named oauth_client_token, token is very generic